adatest/_test_tree.py

```diff
@@ -242,7 +242,7 @@ def append(self, test_tree, axis=0):
         #self.compute_embeddings()
         return None # TODO: Rethink append logic -- return copy vs. in place update?
 
-    def __len__(self):
+    def __len__(self) -> int:
```

Comment:
I think it's best to keep out type hints for the time being (especially on functions like len) to keep a somewhat consistent style in the codebase. Agreed that it's worth revisiting this (and futher mypy support) in the future though!

Reply:
Removed this one - although I would have thought that this was one of the safest type annotations to keep